utils/git.go

```diff
@@ -435,3 +439,36 @@ func setGoGitCustomClient() {
 	client.InstallProtocol("http", githttp.NewClient(customClient))
 	client.InstallProtocol("https", githttp.NewClient(customClient))
 }
+
+// CreateGitInfoContext Creates GitInfoContest for XSC scans,this is optional.
+// only log warning if fails to get information.
+func CreateGitInfoContext(repoName, repoOwner, gitProject string, vcsProvider vcsutils.VcsProvider, client vcsclient.VcsClient, branchName string) (gitInfo *services.XscGitInfoContext) {
+	latestCommit, err := client.GetLatestCommit(context.Background(), repoOwner, repoName, branchName)
+	if err != nil {
```

Comment:
let's return the errors here, and if we decide to just log them currently let's do it only once on the place that calls CreateGitInfoContext.